diff --git a/41_snapshots-btrfs b/41_snapshots-btrfs
index 5e41018..38f1a1d 100755
--- a/41_snapshots-btrfs
+++ b/41_snapshots-btrfs
@@ -45,7 +45,7 @@ prefix="/usr"
 exec_prefix="/usr"
 datarootdir="/usr/share"
 sysconfdir="/etc"
-grub_btrfs_config="${sysconfdir}/default/grub-btrfs/config"
+grub_btrfs_config="${sysconfdir}/default/grub"
 
 [[ -f "$grub_btrfs_config" ]] && . "$grub_btrfs_config"
 . "$datarootdir/grub/grub-mkconfig_lib"
@@ -58,7 +58,7 @@ grub_btrfs_config="${sysconfdir}/default/grub-btrfs/config"
 grub_btrfs_disable=${GRUB_BTRFS_DISABLE:-"false"}
 [[ "${grub_btrfs_disable}" == "true" ]] && exit 0
 ## Submenu name
-submenuname=${GRUB_BTRFS_SUBMENUNAME:-"Arch Linux snapshots"}
+submenuname=${GRUB_BTRFS_SUBMENUNAME:-"Gentoo snapshots"}
 ## Prefix entry
 prefixentry=${GRUB_BTRFS_PREFIXENTRY:-"Snapshot:"}
 ## Show full path snapshot or only name
