diff --git a/Main.qml b/Main.qml
index 6f0d22e..5eab017 100644
--- a/Main.qml
+++ b/Main.qml
@@ -143,7 +143,7 @@ PlasmaCore.ColorScope {
                 },
                 ActionButton {
                     iconSource: "system-shutdown"
-                    text: config.TranslationPowerOff ? config.TranslationPowerOff : i18nd("plasma_lookandfeel_org.kde.lookandfeel","Shutdown")
+                    text: config.TranslationPowerOff ? config.TranslationPowerOff : i18nd("plasma_lookandfeel_org.kde.lookandfeel","Shut Down")
                     onClicked: sddm.powerOff()
                     enabled: sddm.canPowerOff
                     visible: !inputPanel.keyboardActive
